exporters/dsexporter/dsexporter.go

```diff
@@ -1,39 +1,115 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"os"
+	"strings"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
+const service = "https://console-openshift-console.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/k8s/ns/appstudio-grafana/services"
+const check = "prometheus-appstudio-ds"
+var kubeconfig *string
+
 type CustomCollector struct {
-	requestCounter prometheus.Counter
+	konfluxProbeSuccess *prometheus.GaugeVec
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		konfluxProbeSuccess: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "konflux_probe_success",
+			Help: "Availability of konflux service/component",
+		},
+		[]string{"service", "check"}),
 	}
 }
 
 // Describe method sends descriptions of the metrics to Prometheus.
 // When Prometheus scrapes the /metrics endpoint of the exporter,
 // it first calls the Describe method to get a description of all the metrics.
 func (e *CustomCollector) Describe(ch chan<- *prometheus.Desc) {
-	e.requestCounter.Describe(ch)
+	e.konfluxProbeSuccess.Describe(ch)
 }
 
 // Collect method sends the current values of the metrics to Prometheus.
 // After Prometheus understands what metrics are available (using the `Describe` method),
 // it then calls the `Collect` method to actually get the values of those metrics.
 func (e *CustomCollector) Collect(ch chan<- prometheus.Metric) {
-	e.requestCounter.Collect(ch)
+	var availability float64
+
+	availability = IsDataSourceExist(GetDataSources(GetGrafanaResource()), check)
+	e.konfluxProbeSuccess.WithLabelValues(service, check).Set(availability)
+	e.konfluxProbeSuccess.Collect(ch)
+}
+
+// get the grafna resource as a map
+func GetGrafanaResource() map[string]interface{} {
+	clientset := NewKubeClient()
```

Comment:
I think passing the client as an input will make it easier to test.

Reply:
Updated